tests/foreman/ui/test_ldap_authentication.py

```diff
@@ -1254,13 +1117,19 @@ def test_external_new_user_login_and_check_count(session):
             session.browser.refresh()
             count = session.ldapauthentication.read_auth_source_counts('External')
             assert count == current_count + 1
+            # checking delete user can't login anymore
+            delete_rhsso_user(user_details['username'])
```

Comment:
Its good to include a test for this behavior within the test function - but should the call to `delete_rhsso_user` be moved out from the `finally` block that it was in?

Previously, if there was an exception during any of the statements above, `delete_rhsso_user` was called - with this change it only gets called if all the statements above are successful.

If this action is going to be included in the test regardless of an exception, its a change to the test behavior on failure - will this potentially impact other tests in the suite?

Ideally the action is still included as a cleanup - via `pytest.request.addfinalizer`, not via this try/finally pattern.

Reply:
Earlier `delete_rhsso_user(user_details['username'])  ` was not part of the test, but now it is . Let me see if I can add that as you mentioned in pytest.request.addfinalizer.